### ISSUE TITLE
Networking: find better solution for the magic "default" value

### ISSUE THREAD

**Issue by thaJeztah** [docker/docker#19421]
Docker uses a special ("default") value for `--net` if no network is specified.

The "default" value is set as default value in this flag;
https://github.com/docker/docker/blob/73a5393bf3530815a608725fe2546121ccadbfe4/runconfig/opts/parse.go#L83

And was introduced here;
https://github.com/docker/docker/commit/da5a3e6dee80f1f5d4059851e4762ffb0484f7e9

This value was needed because Windows does not have a "bridge" network, so requires some special handling to check if the *default* (type of) network is used, for example, in [`hostconfig_windows.go`](https://github.com/docker/docker/blob/611d5e4eebc4c6153e49a319360df8de5bdd2e7c/vendor/src/github.com/docker/engine-api/types/container/hostconfig_windows.go#L8-L50)

- Should we use a literal "default" string for the default value? Can we use (e.g.) `null` (no value) as the default?
- In the current situation the `--net` flag shows "default" as it's default value, but (on Linux), the default is actually "bridge", so this is confusing

Just opening this for discussion, not a high-priority case.

**Comment by aditirajagopal** [comment]
Can this be closed because of #19431?

**Comment by thaJeztah** [comment]
@aditirajagopal no, that's only to fix a bug; my preference is to not have this special value, but it may need some discussion before that happens

**Comment by mavenugo** [comment]
@thaJeztah I agree. It was added during the "experimental" phase of networking when we wanted to also support the user specified `default` network. But that idea didn't pan out and should have died along with other experimental functionalities that didn't make it.
As you pointed out, windows integration happened around that time and it was carried forward. Now we have a few issues around managing "default" literal. Lets try and remove that in the next release.

**Comment by icecrime** [comment]
I'm assuming this won't make it for 1.11: tentatively removing from milestone.
